tests/core/test_flow.py

```diff
@@ -2524,26 +2525,37 @@ def test_load_accepts_name_and_sluggified_name(self):
 )
 @pytest.mark.parametrize("executor", ["local", "sync", "mthread"], indirect=True)
 def test_timeout_actually_stops_execution(executor):
+    # Note: this is a potentially brittle test! In some cases (local and sync) signal.alarm
+    # is used as the mechanism for timing out a task. This passes off the job of measuring
+    # the time for the timeout to the OS, which uses the "wallclock" as reference (the real
+    # amount of time passed in the real world). However, since the OS balances which processes
+    # can use the CPU and for how long, it is possible when the CPU is strained for the
+    # Python process running the Flow to not be given "enough" time on the CPU after the signal
+    # alarm is registered with the OS. This could result in the Task.run() only percieving a small
+    # amount of CPU time elapsed when in reality the full timeout period had elapsed.
+
+    # For that reason, this test cannot validate timeout functionality by testing "how far into
+    # the task implementation" we got, but instead do a simple task (create a file) and sleep.
+    # This will drastically reduce the brittleness of the test (but not completely).
+
     with tempfile.TemporaryDirectory() as call_dir:
+        # Note: a real file must be used in the case of "mthread"
         FILE = os.path.join(call_dir, "test.txt")
 
         @prefect.task(timeout=1)
         def slow_fn():
-            "Runs for 1.5 seconds, writes to file 7 times"
-            iters = 0
-            while iters < 6:
-                time.sleep(0.25)
-                with open(FILE, "a") as f:
-                    f.write("called\n")
-                iters += 1
+            with open(FILE, "w") as f:
+                f.write("called!")
+            time.sleep(3)
 
         flow = Flow("timeouts", tasks=[slow_fn])
+
+        assert not os.path.exists(FILE)
+
         state = flow.run(executor=executor)
 
-        # if it continued running, would run for 1 more second
-        time.sleep(0.5)
-        with open(FILE, "r") as g:
-            contents = g.read()
+        assert os.path.exists(FILE)
```

Comment:
So with this refactor, the test is no longer testing what it claims to be testing (you're now testing that a timeout occurred but the sleep could still be occurring in the background).  If you think it's too brittle, we might need to consider removing it altogether.

Reply:
I would be OK with removing it, but if timeouts are staying it's a good test to try and keep. I've updated the test to include a footer that should never appear in the file (only added if the task continues).

Reply:
Awesome - good idea!